packages/squiggle-lang/src/rescript/Distributions/GenericDist/GenericDist.res

```diff
@@ -282,6 +312,36 @@ module AlgebraicCombination = {
       }
     }
   }
+
+  let run = (
+    ~strategy: DistributionTypes.asAlgebraicCombinationStrategy,
+    t1: t,
+    ~toPointSetFn: toPointSetFn,
+    ~toSampleSetFn: toSampleSetFn,
+    ~arithmeticOperation,
+    ~t2: t,
+  ): result<t, error> => {
+    switch strategy {
+    | AsDefault => runDefault(t1, ~toPointSetFn, ~toSampleSetFn, ~arithmeticOperation, ~t2)
+    | AsSymbolic =>
+      switch tryAnalyticalSimplification(arithmeticOperation, t1, t2) {
+      | Some(#AnalyticalSolution(symbolicDist)) => Ok(Symbolic(symbolicDist))
+      | Some(#NoSolution)
+      | None =>
+        Error(RequestedStrategyInvalidError)
+      | Some(#Error(err)) => Error(OperationError(err))
+      }
+    | AsConvolution
+    | AsMonteCarlo =>
+      switch chooseConvolutionOrMonteCarlo(~strat=strategy, arithmeticOperation, t1, t2) {
```

Comment:
This should be refactored. It's weird that we have a different method just for choosing chooseConvolutionOrMonteCarlo like this